From 9384c1c0f8b64535595c89bb2b1a980538c8be43 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 6 Mar 2013 13:10:25 +0100
Subject: [PATCH 2/4] Complete quotasync usage
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
Signed-off-by: Jan Kara <jack@suse.cz>
---
 quotasync.c | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/quotasync.c b/quotasync.c
index dd8ade9..cab9015 100644
--- a/quotasync.c
+++ b/quotasync.c
@@ -22,8 +22,23 @@ char *progname;
 
 static void usage(void)
 {
-	errstr(_("Utility for syncing quotas.\nUsage:\n%s [-ug] -a | mntpoint...\n\n"), progname);
-	fprintf(stderr, _("Bugs to %s\n"), MY_EMAIL);
+	printf(_(
+"%1$s: Utility for syncing quotas.\n"
+"Usage: %1$s [-ug] mount-point...\n"
+"   or: %1$s [-ug] -a\n"
+"   or: %1$s -h | -V\n"
+"\n"
+		), progname);
+	printf(_(
+"Options:\n"
+"-u, --user     synchronize user quotas\n"
+"-g, --group    synchronize group quotas\n"
+"-a, --all      synchronize quotas for all mounted file systems\n"
+"-h, --help     display this help message and exit\n"
+"-V, --version  display version information and exit\n"
+"\n"
+		));
+	printf(_("Report bugs to <%s>.\n"), MY_EMAIL);
 	exit(1);
 }
 
-- 
1.8.1.4

